server/api/utils.js

```diff
@@ -4,44 +4,82 @@ import debug from "debug";
 import Router from "koa-router";
 import { AuthenticationError } from "../errors";
 import { Document, Attachment } from "../models";
-import { Op } from "../sequelize";
+import { Op, sequelize } from "../sequelize";
+import parseAttachmentIds from "../utils/parseAttachmentIds";
+import { deleteFromS3 } from "../utils/s3";
 
 const router = new Router();
 const log = debug("utils");
 
 router.post("utils.gc", async (ctx) => {
-  const { token } = ctx.body;
+  const { token, limit = 500 } = ctx.body;
 
   if (process.env.UTILS_SECRET !== token) {
     throw new AuthenticationError("Invalid secret token");
   }
 
-  log("Permanently deleting documents older than 30 days…");
-
-  const where = {
-    deletedAt: {
-      [Op.lt]: subDays(new Date(), 30),
-    },
-  };
+  log(`Permanently destroying upto ${limit} documents older than 30 days…`);
 
   const documents = await Document.scope("withUnpublished").findAll({
-    attributes: ["id"],
-    where,
-  });
-  const documentIds = documents.map((d) => d.id);
-
-  await Attachment.destroy({
+    attributes: ["id", "teamId", "text"],
     where: {
-      documentId: documentIds,
+      deletedAt: {
+        [Op.lt]: subDays(new Date(), 30),
+      },
     },
+    paranoid: false,
+    limit,
   });
 
+  const query = `
```

Comment:
🍖  Find if any other documents reference this attachment, thankfully the searchVector works just fine for this, combined with limiting by `teamId` it's an efficient query.